src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics/PositionDerivative.cpp

```diff
@@ -42,16 +42,22 @@ std::ostream& operator<<(std::ostream& anOutputStream, const PositionDerivative&
     return anOutputStream;
 }
 
+void PositionDerivative::declareCoordinates(const Shared<CoordinatesBroker>& coordinatesBroker)
+{
+    this->positionIndex_ = coordinatesBroker->addSubset(CoordinatesSubset::Position());
+    this->velocityIndex_ = coordinatesBroker->addSubset(CoordinatesSubset::Velocity());
+}
+
 void PositionDerivative::applyContribution(
     const Dynamics::StateVector& x, Dynamics::StateVector& dxdt, const Instant& anInstant
 ) const
 {
     (void)anInstant;
 
     // Integrate position states
-    dxdt[0] = x[3];
-    dxdt[1] = x[4];
-    dxdt[2] = x[5];
+    dxdt[positionIndex_] += x[velocityIndex_];
```

Comment:
```suggestion
    dxdt[positionIndex_] += x[velocityIndex_];
```
Do we really need `+=` here? 🤔